### PR TITLE
asec(0), acos(zoo) is zoo

### PR DESCRIPTION
**Pull request by leosartaj**
According to wolfram `acos(zoo)` and `asec(0)` should be `zoo`. It was also pointed out by @skirpichev in #9223.

### CODE REVIEW

**Comment by leosartaj** [comment]
@jksuom @debugger22

**Comment by jksuom** [comment]
The functions are not defined at the the indicated points. To say that they should have the value ``zoo`` would mean that ``zoo`` were the limit at those points. Since the functions are multiple-valued, it is not clear how the limit should be defined, just for the principal branch or for all branches simultaneously.

I am inclined to believe that the limit in Wolfram is correct, but it would be nice to have some kind of independent proof, perhaps as an addition to the sphinx documents.

**Comment by leosartaj** [comment]
Wolfram uses this [definition](https://reference.wolfram.com/language/ref/ArcSec.html) for complex arguments for asec(x).

```
In [15]: b = -I*(ln(sqrt(1 - x**2) + 1) - ln(x))

In [16]: b.subs(x, 0)
Out[16]: zoo
```
Probably this can support the documentation.
@jksuom

**Comment by jksuom** [comment]
The formula in [15] is a good starting point. On the other hand, I am a bit doubtful of the power of [16] as evidence since it leads to a limit of the form ``zoo - zoo`` when the negative branch of ``sqrt`` is taken.
So, one wood also need the limit of ``ln(-sqrt(1 - x**2) + 1) - ln(x)`` as x tends to 0.

**Comment by asmeurer** [comment]
There are two different definitions for arcsec (at least for the real valued function). I'm not sure which one we use. 

Anyway, if arcsec is multivalued, we have to pick a branch, and assumedly that function in terms of sqrt and ln is chosen so that the principal branches of those two functions correspond to the correct branch for arcsec.

**Comment by jksuom** [comment]
The sources seem to agree that the branch cut of asec is the interval from -1 to 1. In particular, the origin lies on the branch cut. If we choose one branch of ``sqrt(1 - x**2)`` in the upper half plane in the neighbourhood of the origin, then the opposite branch appears in the lower half plane. Hence we have to consider both branches of the square root to find the limit at the origin.

Fortunately, that is not hard to do manually. One can proceed along the following lines:
```
ln(-sqrt(1 - x**2) + 1) - ln(x) = ln((1 - sqrt(1 - x**2))/x)
= ln((1 - (1 - x**2/2 + O(x**4)))/x) = ln(x/2 + O(x**3))
```
with the limit ``zoo`` as x tend to 0. So the limit is the same on both branches and we are justified in using that value.

**Comment by jksuom** [comment]
I seem to have succeeded in convincing myself that these definitions are correct. Still, I would very much like to see some explanations in the docstrings. The expression with log and sqrt should be there. And then the behaviour of ``(+/-sqrt(1 - x**2) + 1)/x`` at the origin. With the + sign we have the limit ``2/0 = zoo``. In case of the - sign one could remark that we get the derivative 0 of ``-sqrt(1 - x**2)`` at the origin, And finally ``log(zoo) = log(0) = zoo``. The behaviour of ``acos`` would follow from ``acos(x) = asec(1/x)``.

**Comment by leosartaj** [comment]
I agree :+1:

**Comment by leosartaj** [comment]
@jksuom ping

File: sympy/functions/elementary/trigonometric.py [hunk 1]
```diff
@@ -2279,8 +2285,26 @@ class asec(InverseTrigonometricFunction):
...
+    .. math::
+        sec^{-1}(z) = -i*(log(\sqrt{1 - z^2} + 1) - log(z))
+
+    At ``x = 0``, for positive branch cut, the limit evaluates to ``zoo``. For
+    negative branch cut, the limit
+
+    .. math::
+        \lim_{z \to 0}-i*(log(-\sqrt{1 - z^2} + 1) - log(z))
+
+    we get the derivative of :math:`-\sqrt(1 - z^2)` which ultimately evaluates
```
Comment (jksuom) [inline sympy/functions/elementary/trigonometric.py]: Actually, we get (-i times) the *logarithm* of the derivative.

Perhaps we should write the definition in the form ``-i*\log((sqrt{1 - z^2} + 1)/z)`` and consider the limits of the two branches of the argument of the logarithm. It is not necessary to copy the exact form from [4].

**Comment by leosartaj** [comment]
I have modified the docstring @jksuom

**Comment by jksuom** [comment]
This looks fine. The two document commits should probably be squashed. (And possibly the first 'which' starting the line "which can be simplified to..." could be omitted.)

**Comment by leosartaj** [comment]
I have squashed the documentation commits and removed the line you suggested @jksuom

**Comment by jksuom** [comment]
There seems to have been some misunderstanding. What I meant was to only remove the extra 'which' starting the line. The rest containing the simplified form with ``x/2 + O(x^3)`` could remain..

**Comment by leosartaj** [comment]
Oops!

**Comment by leosartaj** [comment]
is this fine? @jksuom

**Comment by jksuom** [comment]
Yes. I'll be merging in a few hours after Travis gets done.

**Pull request merged**